source/constructs/api/data_source/service.py

```diff
@@ -1193,41 +1212,221 @@ def before_delete_rds_connection(account: str, region: str, instance: str):
+                        }
+                    },
+                    Permissions=['ALL'],
+                    PermissionsWithGrantOption=['ALL']
+                )
+            except Exception as e:
+                sleep(SLEEP_MIN_TIME)
+                num_retries -= 1
+            else:
+                break
+        else:
+            raise Exception('UNCONNECTED')
+        try:
+            gt_cr_response = glue.get_crawler(Name=crawler_name)
+            logger.info(gt_cr_response)
+            try:
+                if state == ConnectionState.ACTIVE.value or state == ConnectionState.UNSUPPORTED.value \
+                        or state == ConnectionState.ERROR.value or state == ConnectionState.STOPPING.value:
+                    up_cr_response = glue.update_crawler(
+                        Name=crawler_name,
+                        Role=crawler_role_arn,
+                        DatabaseName=glue_database_name,
+                        Targets={
+                            "S3Targets": build_s3_targets(bucket, credentials, region, False)
+                        },
+                        SchemaChangePolicy={
+                            'UpdateBehavior': 'UPDATE_IN_DATABASE',
+                            'DeleteBehavior': 'DELETE_FROM_DATABASE'
+                        }
+                    )
+                    logger.info("update crawler:")
+                    logger.info(up_cr_response)
+            except Exception as e:
+                logger.info("update_crawler s3 error")
+                logger.info(str(e))
+            response = glue.start_crawler(
+                Name=crawler_name
+            )
+            logger.info(response)
+        except Exception as e:
+            response = glue.create_crawler(
+                Name=crawler_name,
+                Role=crawler_role_arn,
+                DatabaseName=glue_database_name,
+                Targets={
+                    "S3Targets": s3_targets
+                },
+                Tags={
+                    'AdminAccountId': _admin_account_id
+                }
+            )
+            logger.info(response)
+            response = glue.start_crawler(
+                Name=crawler_name
+            )
+            logger.info(response)
+
+        crud.create_s3_connection(account, region, bucket, glue_connection_name, glue_database_name, crawler_name)
+    except Exception as err:
+        crud.set_s3_bucket_source_glue_state(account, region, bucket, str(err))
+        glue = boto3.client('glue',
+                            aws_access_key_id=credentials['AccessKeyId'],
+                            aws_secret_access_key=credentials['SecretAccessKey'],
+                            aws_session_token=credentials['SessionToken'],
+                            region_name=region
+                            )
+        """ :type : pyboto3.glue """
+
+        try:
+            glue.delete_crawler(crawler_name)
+        except Exception as e:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Try, Except, Pass detected. https://bandit.readthedocs.io/en/latest/plugins/b110_try_except_pass.html